tests/components/nest/test_config_flow_legacy.py

```diff
@@ -1,193 +1,223 @@
+    )
     assert result["type"] == data_entry_flow.RESULT_TYPE_ABORT
     assert result["reason"] == "missing_configuration"
 
 
 async def test_abort_if_single_instance_allowed(hass):
     """Test we abort if Nest is already setup."""
-    flow = config_flow.NestFlowHandler()
-    flow.hass = hass
+    existing_entry = MockConfigEntry(domain=DOMAIN, data={})
+    existing_entry.add_to_hass(hass)
 
-    with patch.object(hass.config_entries, "async_entries", return_value=[{}]):
-        result = await flow.async_step_init()
+    assert await async_setup_component(hass, DOMAIN, CONFIG)
+    await hass.async_block_till_done()
 
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
     assert result["type"] == data_entry_flow.RESULT_TYPE_ABORT
     assert result["reason"] == "single_instance_allowed"
 
 
 async def test_full_flow_implementation(hass):
     """Test registering an implementation and finishing flow works."""
-    gen_authorize_url = AsyncMock(return_value="https://example.com")
-    convert_code = AsyncMock(return_value={"access_token": "yoo"})
-    config_flow.register_flow_implementation(
-        hass, "test", "Test", gen_authorize_url, convert_code
-    )
+    assert await async_setup_component(hass, DOMAIN, CONFIG)
+    await hass.async_block_till_done()
+    # Register an additional implementation to select from during the flow
     config_flow.register_flow_implementation(
         hass, "test-other", "Test Other", None, None
     )
 
-    flow = config_flow.NestFlowHandler()
-    flow.hass = hass
-    result = await flow.async_step_init()
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
     assert result["type"] == data_entry_flow.RESULT_TYPE_FORM
     assert result["step_id"] == "init"
 
-    result = await flow.async_step_init({"flow_impl": "test"})
+    result = await hass.config_entries.flow.async_configure(
+        result["flow_id"],
+        {"flow_impl": "nest"},
+    )
     assert result["type"] == data_entry_flow.RESULT_TYPE_FORM
     assert result["step_id"] == "link"
-    assert result["description_placeholders"] == {"url": "https://example.com"}
+    assert (
+        result["description_placeholders"]
+        .get("url")
+        .startswith("https://home.nest.com/login/oauth2?client_id=bla")
+    )
 
-    result = await flow.async_step_link({"code": "123ABC"})
-    assert result["type"] == data_entry_flow.RESULT_TYPE_CREATE_ENTRY
-    assert result["data"]["tokens"] == {"access_token": "yoo"}
-    assert result["data"]["impl_domain"] == "test"
-    assert result["title"] == "Nest (via Test)"
+    def mock_login(auth):
+        assert auth.pin == "123ABC"
+        auth.auth_callback({"access_token": "yoo"})
+
+    with patch(
+        "homeassistant.components.nest.legacy.local_auth.NestAuth.login", new=mock_login
+    ):
```

Comment:
Should we patch out `homeassistant.components.nest.async_setup_entry` here as well?

Reply:
Done.  I also made sure this calls into `async_setup_entry` and calls `async_setup_legacy` to make sure the wiring still works, and I updated the call below as well.

Reply:
Take another look just to be sure I understood your comment right.